taichi/program/async_engine.cpp

```diff
@@ -210,10 +210,11 @@ void AsyncEngine::enqueue(const TaskLaunchRecord &t) {
 void AsyncEngine::synchronize() {
   TI_AUTO_PROF
   bool modified = true;
-  TI_TRACE("Synchronizing SFG of {} nodes", sfg->size());
-  debug_sfg("initial");
   sfg->reid_nodes();
   sfg->reid_pending_nodes();
+  TI_TRACE("Synchronizing SFG of {} nodes ({} pending)", sfg->size(),
+           sfg->num_pending_tasks());
+  debug_sfg("initial");
```

Comment:
Off-topic change: `num_pending_tasks()` matches the following `Ended up with {} nodes` better.